src/java/org/apache/cassandra/transport/CQLMessageHandler.java

```diff
@@ -163,7 +163,7 @@ protected boolean processOneContainedMessage(ShareableBytes bytes, Limit endpoin
         
         if (throwOnOverload)
         {
-            if (DatabaseDescriptor.getNativeTransportRateLimitingEnabled() && !requestRateLimiter.tryReserve())
+            if (DatabaseDescriptor.getNativeTransportRateLimitingEnabled() && requestRateLimiter.reserveAndGetWaitMicros() > 0)
```

Comment:
I think you've replaced a conditional reservation with an unconditional one, so that these overloads still count towards the number of requests we're processing?